frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx

```diff
@@ -0,0 +1,484 @@
+    ? 'The required custom resource will need to be created in the operator once resolved.'
+    : '';
+  const installMessage = `The operator did not install successfully. ${crdMessage}`;
+
+  return (
+    <>
+      <span className="co-clusterserviceversion-install__heading">
+        Operator installation failed
+      </span>
+      <br />
+      <span>{installMessage}</span>
+      <div className="co-clusterserviceversion-install__button_div">
+        <Button
+          variant="primary"
+          component="a"
+          href={`/k8s/ns/${namespace}/${referenceForModel(ClusterServiceVersionModel)}/${csvName}`}
+        >
+          View Error
+        </Button>{' '}
+        <ViewInstalledOperatorsButton namespace={namespace} className={''} />
+      </div>
+    </>
+  );
+};
+
+const InstallNeedsApprovalMessage: React.FC<InstallPageMessageProps> = ({
+  namespace,
+  obj,
+  approve,
+}) => (
+  <>
+    <span className="co-clusterserviceversion-install__heading">Manual approval required</span>
+    <br />
+    <p>
+      Review the manual install plan. Once approved, the following resources will be created in
+      order to satisfy the requirements for the components specified in the plan. Click the resource
+      name to view the resource in detail.
+    </p>
+    <div className="co-clusterserviceversion-install__button_div">
+      <Button variant="primary" onClick={() => approve()}>
+        Approve
+      </Button>
+      <Button
+        className="co-clusterserviceversion-button__left_padding"
+        variant="secondary"
+        component="a"
+        href={`/k8s/ns/${namespace}/${referenceForModel(SubscriptionModel)}/${
+          obj.metadata.name
+        }?showDelete=true`}
+      >
+        Deny
+      </Button>
+      <ViewInstalledOperatorsButton namespace={namespace} className={''} />
+    </div>
+  </>
+);
+
+const CreateCRDButton: React.FC<InstallPageMessageButtonProps> = ({ className, obj, disabled }) => {
+  if (!isCRDAvailable(obj)) {
+    return <></>;
+  }
+  const crObj = safeLoad(getCRDFromObject(obj));
+  const crdNamespace = crObj?.[0]?.metadata?.namespace;
+  const apiVersion = crObj?.[0]?.apiVersion;
+  const kind = crObj?.[0]?.kind;
+  const apiLink = referenceForGroupVersionKind(groupVersionFor(apiVersion).group)(
+    groupVersionFor(apiVersion).version,
+  )(kind);
+  return (
+    <Button
+      isDisabled={disabled}
+      className={className}
+      variant="primary"
+      component="a"
+      href={`/k8s/ns/${crdNamespace}/${referenceForModel(ClusterServiceVersionModel)}/${
+        obj?.metadata?.name
+      }/${apiLink}/~new`}
+    >
+      {`Create ${kind}`}
+    </Button>
+  );
+};
+
+const InstallPageCRDRequiredMessage: React.FC<InstallPageMessageProps> = ({ obj }) => {
+  if (!isCRDAvailable(obj)) {
+    return <></>;
+  }
+  const crObj = safeLoad(getCRDFromObject(obj));
+
+  const crdKind = crObj?.[0]?.kind;
+  const crdNameSpace = crObj?.[0]?.metadata?.namespace;
```

Comment:
Namespace is one word

```suggestion
  const crdNamespace = crObj?.[0]?.metadata?.namespace;
```

This name is a bit confusing though since CRDs are not namespaced. I assume this is the namespace for the custom resource.